### PR TITLE
Fix TextRank algorithm

### PR DESCRIPTION
**Pull request by kmkurn**
This PR fixes the TextRank algorithm so that it runs PageRank algorithm (as discussed in #87). This PR also:

- Adds `requirements.txt` file so contributors can install all the required dependencies easily. I don't know the exact version that you use, so maybe you can check if all the versions are correct.
- Updates pytest config section name in `setup.cfg` to `[tool:pytest]` instead of `[pytest]`; the latter is deprecated.
- Updates README by referencing the original TextRank paper. (I notice there is a README.rst file as well, why the redundancy? Github can parse .rst file just fine.)
- Fixes edge case when computing sentence similarity in TextRank. This guarantees the adjacency matrix has non-zero values in the diagonals, which makes more sense because a sentence should be similar to itself.

Some things to note:

- Current version of pytest does not seem to have pytest executable for every python version. I believe [tox](https://pypi.python.org/pypi/tox) is now the standard tool to do tests on several python versions.
- If this PR is accepted, `.travis.yml` file might be modified so it installs dependencies from `requirements.txt` so there's only one place where dependencies are listed.

### CODE REVIEW

**Comment by miso-belica** [comment]
@kmkurn Thanks for this. From the brief view. Can you please remove file `requirements.txt`? sumy is library and not application and I don't want to rely on specific versions. Dependencies should be installed from `setup.py` file. See https://caremad.io/posts/2013/07/setup-vs-requirement/ for explanation.

Also I don't like you removed old algorithm. Can you please create new class for TextRank algorithm and let the old summarization algorithm in sumy with some other name (for example `ReductionSummarizer`) so it will be still available? Some people like ti and use it :)

**Comment by kmkurn** [comment]
1) Yes sumy is a library, but `requirements.txt` file's purpose is for the developers/contributors, not users. For the developers, sumy becomes an application (this is mentioned in the article you linked), so ideally they all have the same dev environment which includes the exact Python and library versions. Developers do not install sumy from pypi but rather they clone the repository and install from `requirements.txt`. Projects like nltk and requests also have `requirements.txt` file (or similar). I can remove it if you insist though.

2) Sure no problem.

File: setup.cfg [hunk 1]
```diff
@@ -7,7 +7,7 @@ tag = false
 
 [bumpversion:file:sumy/__init__.py]
 
-[pytest]
+[tool:pytest]
 addopts = --quiet --tb=short --color=yes --cov=sumy --cov-report=term-missing --no-cov-on-fail
```
Comment (kmkurn) [inline setup.cfg]: @miso-belica you have pytest-cov options listed here, so anyone invoking `pytest` command (without any options) will have to have pytest-cov installed. So it is not optional. Thus, it's either removing pytest-cov options from here or make it required as a dependency.

Comment (miso-belica) [reply setup.cfg]: @kmkurn Thanks for this. Then make pytest-cov required please.

**Comment by kmkurn** [comment]
@miso-belica I've implemented the requested changes but the build failed for Python 3.7 and nightly when doing `python setup.py install`. It worked just find for earlier version of Python so I'm not sure why. Do you have any idea?

**Comment by miso-belica** [comment]
@kmkurn thanks, I think you can ignore it now. I will try to figure it later or just will ignore it because it's dev versions anyway, now I am enjoying The PyCon ☺️

**Pull request merged**